app/controllers/products_controller.rb

```diff
@@ -1,5 +1,34 @@
 class ProductsController < ApplicationController
+
   def index
     @products = Product.all
   end
+
+  before_action :authenticate_admin!, only: [:edit, :update, :destroy]
```

Comment:
remove everything in this file that is out of scope for this feature

Reply:
Done :)